include/boost/geometry/util/series_expansion.hpp

```diff
@@ -0,0 +1,746 @@
+            break;
+        case 4:
+            c[1] = d*(eps2+CT(8))/CT(16);
+            d *= eps;
+            c[2] = d*(eps2+CT(6))/CT(32);
+            d *= eps;
+            c[3] = CT(5)*d/CT(48);
+            d *= eps;
+            c[4] = CT(35)*d/CT(512);
+            break;
+        case 5:
+            c[1] = d*(eps2*(eps2+CT(2))+CT(16))/CT(32);
+            d *= eps;
+            c[2] = d*(eps2+CT(6))/CT(32);
+            d *= eps;
+            c[3] = d*(CT(15)*eps2+CT(80))/CT(768);
+            d *= eps;
+            c[4] = CT(35)*d/CT(512);
+            d *= eps;
+            c[5] = CT(63)*d/CT(1280);
+            break;
+        case 6:
+            c[1] = d*(eps2*(eps2+CT(2))+CT(16))/CT(32);
+            d *= eps;
+            c[2] = d*(eps2*(CT(35)*eps2+CT(64))+CT(384))/CT(2048);
+            d *= eps;
+            c[3] = d*(CT(15)*eps2+CT(80))/CT(768);
+            d *= eps;
+            c[4] = d*(CT(7)*eps2+CT(35))/CT(512);
+            d *= eps;
+            c[5] = CT(63)*d/CT(1280);
+            d *= eps;
+            c[6] = CT(77)*d/CT(2048);
+            break;
+        case 7:
+            c[1] = d*(eps2*(eps2*(CT(41)*eps2+CT(64))+CT(128))+CT(1024))/CT(2048);
+            d *= eps;
+            c[2] = d*(eps2*(CT(35)*eps2+CT(64))+CT(384))/CT(2048);
+            d *= eps;
+            c[3] = d*(eps2*(CT(69)*eps2+CT(120))+CT(640))/CT(6144);
+            d *= eps;
+            c[4] = d*(CT(7)*eps2+CT(35))/CT(512);
+            d *= eps;
+            c[5] = d*(CT(105)*eps2+CT(504))/CT(10240);
+            d *= eps;
+            c[6] = CT(77)*d/CT(2048);
+            d *= eps;
+            c[7] = CT(429)*d/CT(14336);
+            break;
+        default:
+            c[1] = d*(eps2*(eps2*(CT(41)*eps2+CT(64))+CT(128))+CT(1024))/CT(2048);
+            d *= eps;
+            c[2] = d*(eps2*(eps2*(CT(47)*eps2+CT(70))+CT(128))+CT(768))/CT(4096);
+            d *= eps;
+            c[3] = d*(eps2*(CT(69)*eps2+CT(120))+CT(640))/CT(6144);
+            d *= eps;
+            c[4] = d*(eps2*(CT(133)*eps2+CT(224))+CT(1120))/CT(16384);
+            d *= eps;
+            c[5] = d*(CT(105)*eps2+CT(504))/CT(10240);
+            d *= eps;
+            c[6] = d*(CT(33)*eps2+CT(154))/CT(4096);
+            d *= eps;
+            c[7] = CT(429)*d/CT(14336);
+            d *= eps;
+            c[8] = CT(6435)*d/CT(262144);
+            break;
+        }
+    }
+
+    /*
+     The coefficients C3[l] in the Fourier expansion of B3.
+
+     The expansion below is performed in Maxima, a Computer Algebra System.
+     The C++ code (that yields the function evaluate_coeffs_C3 below) is
+     generated by the following Maxima script:
+     geometry/doc/other/maxima/geod.mac
+    */
+    template <typename Coeffs, typename CT>
+    inline void evaluate_coeffs_C3x(Coeffs &c, size_t const& SeriesOrder, CT const& n) {
+        size_t const coeff_size = Coeffs::static_size;
+        BOOST_GEOMETRY_ASSERT(coeff_size == (SeriesOrder * (SeriesOrder - 1)) / 2);
```

Comment:
If you took `SeriesOrder` as template parameter you could check this at compile time:

    static size_t const coeff_size = Coeffs::static_size;
    BOOST_MPL_ASSERT(((coeff_size == (SeriesOrder * (SeriesOrder - 1)) / 2)));

Reply:
I'm having some trouble passing `SeriesOrder` as the template parameter.

I changed the `evaluate_coeffs_C3x` function header to:
```cpp
template <typename Coeffs, typename CT, size_t SeriesOrder>
inline void evaluate_coeffs_C3x(Coeffs &c, CT const& n) {
```
and called it as:
```cpp
evaluate_coeffs_C3x(*this, n);
```
but I get this error:
```
error: ‘evaluate_coeffs_C3x’ was not declared in this scope
```
I also tried changing the order of parameters as:
```cpp
template <size_t SeriesOrder, typename CT, typename Coeffs>
inline void evaluate_coeffs_C3x(CT const& n, Coeffs &c) {
```
but calling the function as:
```cpp
evaluate_coeffs_C3x(n, *this);
```
or:
```cpp
evaluate_coeffs_C3x<SeriesOrder, CT>(n, *this);
```
doesn't work either.

It produces the error:
```
error: expected primary-expression before ‘>’ token
```
and:
```
couldn't deduce template parameter ‘SeriesOrder’
```
respectively.

Reply:
Either of these should work:

    template <size_t SeriesOrder, typename CT, typename Coeffs>
    inline void evaluate_coeffs_C3x(CT const& n, Coeffs &c){}

    // this is more strict, it takes only the struct you like
    template <size_t SeriesOrder, typename CT>
    inline void evaluate_coeffs_C3x(CT const& n, coeffs_C3x<SeriesOrder, CT> &c){}

and either of these should work:

    evaluate_coeffs_C3x<SeriesOrder>(n, *this);
    evaluate_coeffs_C3x<SeriesOrder, CT>(n, *this);
    evaluate_coeffs_C3x<SeriesOrder, CT, coeffs_C3x>(n, *this); // assuming it's called in coeffs_C3x's constructor
    
If you're calling these functions inside structs' constructors then they should be defined before structs.

Reply:
Thanks, this worked.

Out of curiosity, why was this working before for the previous function calls? Does this have something to do with passing explicit template parameters? As I only got the error with `evaluate_coeffs_C3x<SeriesOrder>`.

Reply:
No idea. You'd have to know how exactly type deduction is done in your compiler. Are you using Visual Studio? Its compiler allows things it shouldn't sometimes.

Reply:
No, I'm using `g++` version `7.2.0` on Ubuntu.